modules/Sources/Dependencies/FiatPriceClient/FiatPriceClient.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  FiatPriceClient.swift
+//
+//
+//  Created by Matthew Watt on 10/1/23.
+//
+
+import ComposableArchitecture
+import Foundation
+import Models
+import Utils
+
+public struct FiatPriceClient {
+    public var getZcashPrice: (NighthawkSetting.FiatCurrency) async throws -> Double?
+}
+
+extension FiatPriceClient: DependencyKey {
+    struct FiatPriceApiResponse: Codable, Equatable {
+        let data: [String: Double]
+        
+        enum CodingKeys: String, CodingKey {
+            case data = "zcash"
+        }
+    }
+    
+    public static let liveValue = Self(
```

Comment:
I'd consider wrapping this up into a caching client that can determine when it is appropriate to refresh the value or return what's already cached. You could also add a setting in your awesome settings screen to adjust the "refresh" time, so that you can let the user decide how often to refresh the cache.

Reply:
See my comment above. Normally you'd be right and yes, we'd have some sort of intelligent caching scheme, but in this case, we just want to always fetch this value if it isn't present in-mem, and discard it when the app's memory is reclaimed.